### ISSUE TITLE
Memory leak in logger

### ISSUE THREAD

**Issue by chep** [mavlink/MAVSDK#2232]
I tried to download files via FTP with MAVSDK_FTP_DEBUGGING and MAVSDK_MESSAGE_DEBUGGING enabled.

The memory consumption was really big and always increased until OOM. I think there is a memory leak in `Log*` functions because when I disable debug, the problem is gone.

**Comment by julianoes** [comment]
Interesting! That will be fun to track down. Thanks for the report.

**Comment by julianoes** [comment]
Could the leak be in the FTP code? What sort of FTP action are you doing when debugging this?

**Comment by chep** [comment]
I was using ftp client with burst download.

**Comment by julianoes** [comment]
I don't think this is still an issue after quite some changes. Let me know if I'm wrong.
